ec/src/hashing/field_hashers/mod.rs

```diff
@@ -1,106 +1,106 @@
+mod expander;
 use ark_ff::{Field, PrimeField};
+
+use crate::hashing::{
+    field_hashers::expander::ExpanderXmd, map_to_curve_hasher::*, HashToCurveError,
+};
 use ark_std::vec::Vec;
+use digest::DynDigest;
+use expander::Expander;
+
+/// This field hasher constructs a Hash-To-Field based on a fixed-output hash function,
+/// like SHA2, SHA3 or Blake2.
+/// The implementation aims to follow the specification in [Hashing to Elliptic Curves (draft)](https://tools.ietf.org/pdf/draft-irtf-cfrg-hash-to-curve-13.pdf).
+///
+/// # Examples
+///
+/// ```
+/// use ark_test_curves::bls12_381::Fq;
+/// use ark_ec::hashing::field_hashers::DefaultFieldHasher;
+/// use sha2::Sha256;
+/// use crate::ark_ec::hashing::map_to_curve_hasher::HashToField;
+///
+/// let hasher = <DefaultFieldHasher<Sha256, 128> as HashToField<Fq>>::new_hash_to_field(&[1, 2, 3]).unwrap();
+/// let field_elements: Vec<Fq> = hasher.hash_to_field(b"Hello, World!", 2).unwrap();
+///
+/// assert_eq!(field_elements.len(), 2);
+/// ```
+pub struct DefaultFieldHasher<H: Default + DynDigest + Clone, const SEC_PARAM: usize> {
```

Comment:
Do we want to have a default `SEC_PARAM`?  Default const generics are stable since 1.59.

Reply:
Yeah, probably makes sense. Will adapt